shared_model/interfaces/commands/add_peer.hpp

```diff
@@ -0,0 +1,48 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_SHARED_MODEL_ADD_PEER_HPP
+#define IROHA_SHARED_MODEL_ADD_PEER_HPP
+
+#include "interfaces/common_objects/types.hpp"
+#include "interfaces/primitive.hpp"
+#include "model/commands/add_peer.hpp"
+
+namespace shared_model {
+  namespace interface {
+
+    /**
+     * Add new peer to Iroha
+     */
+    class AddPeer : public Primitive<AddPeer, iroha::model::AddPeer> {
```

Comment:
All command data interfaces should be inherited from `Hashable` because they contain in `Signable` object.

This issue related to other commands.